framework/Steps.py

```diff
@@ -427,16 +434,34 @@ def _localInitializeStep(self,inDictionary):
     if inDictionary['Model'].createWorkingDir:
       currentWorkingDirectory = os.path.join(inDictionary['jobHandler'].runInfoDict['WorkingDir'],
                                              inDictionary['jobHandler'].runInfoDict['stepName'])
-      try:
-        os.mkdir(currentWorkingDirectory)
-      except FileExistsError:
-        self.raiseAWarning('current working dir '+currentWorkingDirectory+' already exists, ' +
-                           'this might imply deletion of present files')
-        if utils.checkIfPathAreAccessedByAnotherProgram(currentWorkingDirectory,3.0):
-          self.raiseAWarning('directory '+ currentWorkingDirectory + ' is likely used by another program!!! ')
-        if utils.checkIfLockedRavenFileIsPresent(currentWorkingDirectory,self.lockedFileName):
-          self.raiseAnError(RuntimeError, self, "another instance of RAVEN is running in the working directory "+ currentWorkingDirectory+". Please check your input!")
-        # register function to remove the locked file at the end of execution
+      workingDirReady = False
+      alreadyTried = False
+      while not workingDirReady:
+        try:
+          os.mkdir(currentWorkingDirectory)
+          workingDirReady = True
+        except FileExistsError:
+          if self._clearRunDir and not alreadyTried:
+            self.raiseAWarning(f'The calculation run directory {currentWorkingDirectory} already exists, ' +
+                              'clearing existing files. This action can be disabled through the RAVEN Step input.')
+
+            utils.removeDir(currentWorkingDirectory)
+            alreadyTried = True
```

Comment:
Why do we need this "try and repeat" approach? When can this happen?

Reply:
I didn't want to copy the code for writing the directory, and I wanted it to follow the same checking logic and making the directory logic as before.

This was the smallest-impact way I could think of to avoid the code duplication.

Reply:
instead of the `try ... except` we could check if the dir exists and build the tree off that `if` instead of the `try`, but I think that's slightly error prone in python in case the directory is modified between the check and the end of the conditional, iirc.